### PR TITLE
Fixed program logging issues

### PR DESCRIPTION
**Pull request by vadim0x60**
Such as:
- A long is_already_solved check when the program doesn't even exist yet
- solutions_dir not being found due to current workdir schenanigans

### CODE REVIEW

File: seidr/github.py
Comment (Anastasiia-Grishina) [inline seidr/github.py:1]: Line 39 in `github.py`: why don't we log a link to the corresponding commit?

```
    commit_url = repo.remotes.origin.url + "/commit/" + repo.head.commit.hexsha
    commit_url = 'https://' + commit_url[commit_url.find('@github.com') + 1:]
    logging.info(f'Tried to push updates to git repo.\n'
                 f'See commit at {commit_url}. \n'
                 f'Commit message: {message}')
```

**Review by Anastasiia-Grishina — CHANGES_REQUESTED** [review]
Great work :raised_hands:

I added one comment regarding logging the commit url. This may be interesting for faster log analysis and cherry-picking examples for the paper. If you could incorporate it, we are ready to go, from my side!

**Review by Anastasiia-Grishina — APPROVED** [review]

File: seidr/github.py
Comment (Anastasiia-Grishina) [reply seidr/github.py:1]: In the interest of time, I'll do this in a separate branch :v:

Comment (vadim0x60) [reply seidr/github.py:1]: Looks good to me!

**Pull request merged**